src/main.rs

```diff
@@ -153,12 +206,24 @@ fn work() -> Result<(), Cue2CCDError> {
         return Err(Cue2CCDError::MissingFilesError { missing_files });
     }
 
+    let mut sbi_hash_map: HashMap<i64, Vec<u8>> = HashMap::new();
+
+    // TODO: is this extension check case sensitive?
+    if Path::new(&output_stem.with_extension("sbi")).exists() {
+        // SBI files are very small, so it seems best to read the whole thing in first?
+        let temp_hashmap = generate_sbi_data(std::fs::read(Path::new(
+            &output_stem.with_extension("sbi"),
+        ))?)?;
+        sbi_hash_map = temp_hashmap;
+    }
+
     let sub_target = output_stem.with_extension("sub");
     let mut sub_write = File::create(sub_target)?;
 
     let disc = Disc::from_cuesheet(cd, root);
     for sector in disc.sectors() {
-        sub_write.write_all(&sector.generate_subchannel(&chosen_protection_type))?;
+        sub_write
+            .write_all(&sector.generate_subchannel(&chosen_protection_type, Some(&sbi_hash_map)))?;
```

Comment:
Huh, now that I think about it... maybe this doesn't even need to be an option after all? If it's empty, then we just generate the image as normal. That feels like it could simplify things a bit.

With the other format you were considering parsing in addition to SBI, is the data we're parsing out of it basically going to be the same thing? So the value passed in to this one param will be basically the same thing, just its source is SBI or something else?

Reply:
I'm going to make a different parameter for LSD, but otherwise the data we're going to be parsing out of it is the same thing.
If for whatever reason someone has both an LSD and an SBI file, I'll just use the LSD file and ignore the SBI file.